hibernate-core/src/main/java/org/hibernate/event/internal/DefaultDeleteEventListener.java

```diff
@@ -263,6 +263,7 @@ private boolean canBeDeletedWithoutLoading(EventSource source, EntityPersister p
 			&& !persister.hasSubclasses() //TODO: should be unnecessary, using EntityPersister.getSubclassPropertyTypeClosure(), etc
 			&& !persister.hasCascadeDelete()
 			&& !persister.hasNaturalIdentifier()
+			&& !persister.hasNaturalIdCollections()
```

Comment:
Not sure about the name here.  I looked through how this comes to be and I think its not really related to what the feature of mapping a natural-id.  If I understand correctly, this is merely indicating whether the entity contains any collection where the collection-key does not refer to the entity's identifier.  

Assuming that understanding is correct, I think the term is a bit confusing.

Reply:
yes  you are right, `hasNonPKCollection` may be a little better but still horrible and probably also confusing
or `hasCollectionNotReferencingPK`. Any suggestion is more than welcome :P

Reply:
`hasCollectionNotReferencingPK` sounds reasonable

Reply:
method name changed, thanks